T/TOPCOM/build_tarballs.jl

```diff
@@ -0,0 +1,99 @@
+    # sed -i '/_la_LDFLAGS/ s/$/ -no-undefined/g' lib-src/Makefile.am lib-src-reg/Makefile.am
+    libopts="--enable-static --disable-shared"
+else
+    libopts="--enable-shared --disable-static"
+fi
+autoreconf -vi
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} CPPFLAGS="-I${includedir}/cddlib -I${includedir}" $libopts
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms(;experimental=true))
+platforms_win = filter(Sys.iswindows, platforms)
+platforms_unix = filter(!Sys.iswindows, platforms)
+
+# The products that we will ensure are always built
+products_win = [
+    ExecutableProduct("chiro2allfinetriangs", :chiro2allfinetriangs),
+    ExecutableProduct("chiro2ntriangs", :chiro2ntriangs),
+    ExecutableProduct("points2nflips", :points2nflips),
+    ExecutableProduct("chiro2mintriang", :chiro2mintriang),
+    ExecutableProduct("points2facets", :points2facets),
+    ExecutableProduct("chiro2cocircuits", :chiro2cocircuits),
+    ExecutableProduct("chiro2placingtriang", :chiro2placingtriang),
+    ExecutableProduct("points2finetriangs", :points2finetriangs),
+    ExecutableProduct("cocircuits2facets", :cocircuits2facets),
+    ExecutableProduct("chiro2alltriangs", :chiro2alltriangs),
+    ExecutableProduct("points2nallfinetriangs", :points2nallfinetriangs),
+    ExecutableProduct("chiro2finetriangs", :chiro2finetriangs),
+    ExecutableProduct("points2ntriangs", :points2ntriangs),
+    ExecutableProduct("points2placingtriang", :points2placingtriang),
+    ExecutableProduct("chiro2dual", :chiro2dual),
+    ExecutableProduct("points2flips", :points2flips),
+    ExecutableProduct("santos_dim4_triang", :santos_dim4_triang),
+    ExecutableProduct("chiro2nalltriangs", :chiro2nalltriangs),
+    ExecutableProduct("points2volume", :points2volume),
+    ExecutableProduct("points2finetriang", :points2finetriang),
+    ExecutableProduct("chiro2circuits", :chiro2circuits),
+    ExecutableProduct("points2allfinetriangs", :points2allfinetriangs),
+    ExecutableProduct("points2nalltriangs", :points2nalltriangs),
+    ExecutableProduct("santos_22_triang", :santos_22_triang),
+    ExecutableProduct("points2nfinetriangs", :points2nfinetriangs),
+    ExecutableProduct("chiro2nallfinetriangs", :chiro2nallfinetriangs),
+    ExecutableProduct("chiro2nfinetriangs", :chiro2nfinetriangs),
+    ExecutableProduct("chiro2finetriang", :chiro2finetriang),
+    ExecutableProduct("chiro2triangs", :chiro2triangs),
+    ExecutableProduct("points2alltriangs", :points2alltriangs),
+    ExecutableProduct("points2chiro", :points2chiro),
+    ExecutableProduct("points2triangs", :points2triangs),
+    ExecutableProduct("santos_triang", :santos_triang)
+]
+
+products_unix = [
+    LibraryProduct("libTOPCOM", :libTOPCOM),
+    LibraryProduct("libCHECKREG", :libCHECKREG),
+    products_win...
+]
```

Comment:
I'm slightly unhappy about having different products on different platforms, this can be confusing for users/package developers who'd expect to find `libTOPCOM` everywhere if `TOPCOM_jll.is_available()` is `true`.  I thought about including products also in https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/174, but it looked error-prone to me.  What do you think?

Reply:
We could leave out these products for all platforms or just remove the mingw platforms altogether, both would be fine for me.
But I really don't want to reorganize the source for these libraries, and currently both need symbols from the other one.

Reply:
We might also split this up into a `libTOPCOM_jll` which has just the shared libraries for unix platforms and TOPCOM_jll for all platforms with the binaries (which then might use platform-dependent dependencies ....). But this seems to be a little bit too much effort.

Reply:
> But this seems to be a little bit too much effort.

Agreed.  Ok, well, let's merge this as it is and let's see how it goes!  Thanks everybody!